src/gameengine/collisiondetection/tree/Tree.java

```diff
@@ -133,17 +134,20 @@ public void entityUpdated(double timeToCheck, Entity entity) {
     }
 
     protected void collideShapes(Collision result, double timeToCheck, Entity a, Entity b) {
-        if ((world.getCollisionGroups()[a.getEntityType()] & b.getEntityTypeBitMask()) != 0) {
-            Collision temp = world.getTempCollision();
-            temp.setNoCollision(); // TODO might not need to do this because collideShapes
-            // overwrites temp anyway
-            Shape.collideShapes(a.getShape(), b.getShape(), timeToCheck, temp);
-            if (temp.getCollisionTime() < result.getCollisionTime() - timeInTree) {
-                assert temp.getCollisionTime() <= timeToCheck : "too long" + temp
-                        .getCollisionTime() + ", " + timeToCheck;
-                result.set(temp);
-                result.setCollisionTime(result.getCollisionTime() + timeInTree);
-            }
+        if ((world.getCollisionGroups()[a.getEntityType()] & b.getEntityTypeBitMask()) == 0 || a
+                instanceof RegionSensor && ((RegionSensor) a).containsEntity(b) || b instanceof
+                RegionSensor && ((RegionSensor) b).containsEntity(a)) {
```

Comment:
So if the first condition (& bitmask) == 0 then we know that a and b don't interact and return right away.  If we keep checking the condition that means that the first condition is false so they should interact which means that the region sensors want to be aware of each other.  If the one of the sensors already contains the other then they are already aware.

But... This does point out a defect in a different part of the code where I add an entity to the region sensor as I should add each other to the other (whereas right now only 1 region sensor is  notified).